pkgs/by-name/me/megapixels/package.nix

```diff
@@ -54,6 +54,40 @@ stdenv.mkDerivation (finalAttrs: {
     zbar
   ];
 
+  patches = [
+    /*
+      Megapixels takes a picture and then pipes the raw sensor data to a postprocessing script/program to create an image file with a "normal" encoding like JPEG.
+
+      By default, Megapixels searches for `postprocess.sh` in the following locations:
+      - current working directory (`./data/postprocess.sh`)
+      - `$XDG_CONFIG_HOME/megapixels/postprocess.sh`
+      - SYSCONFDIR (`/etc/megapixels/postprocess.sh`)
+      - DATADIR (`/usr/share/megapixels/postprocess.sh` on non-NixOS and `/nix/store/hash-megapixels-version/share/megapixels/postprocess.sh` on NixOS)
+
+      The Megapixels package itself ships with a postprocessing script in `/nix/store/hash-megapixels-version/share/megapixels/postprocess.sh`.
+
+      Megapixels also searches for extra postprocessor scripts in DATADIR (`/usr/share/megapixels/postprocessor.d/` on non-NixOS and `/nix/store/hash-megapixels-version/share/megapixels/postprocessor.d/` on NixOS). This file does not exist on NixOS because Megapixels does not ship with any extra postprocessor scripts.
+
+      This patch modifies Megapixels to also search for `postprocess.sh` at the path `/run/current-system/sw/share/megapixels/postprocess.sh` and to search for `postprocessor.d/` in `/run/current-system/sw/share/megapixels/postprocessor.d/` instead of `/nix/store/hash-megapixels-version/share/megapixels/postprocessor.d/` as the latter path never exists.
+      This makes Megapixels behave more like it does on other distributions where one can install other postprocessors like `postprocessd` globally and have Megapixels be able to use these extra postprocessors.
+
+      It would be nicer to develop some kind of plugin system that aligns more with the NixOS philosophy.
+      This would allow:
+      - Running Megapixels from a nix shell and have access to extra postprocessing scripts instead of having to install globally
+      - Installing Megapixels for different users with different postprocessing scripts installed per user without requiring hacks like setting the `/org/postmarketos/megapixels/postprocessor` dconf setting manually
+
+      but I (@Luflosi) won't do that as:
+      - I don't know how to do it
+      - This is much simpler
+      - In the settings menu of Megapixels the user can select a different postprocessing script. The path to the script is then stored in a dconf setting. If the path changes, for example because it is in the Nix store and a dependency of Megapixels or the postprocessor changes, Megapixels will try to use this now non-existing old path. This will cause Megapixels to not save any images that were taken until the user opens the settings again and selects a postprocessor again. Using a global path allows the setting to keep working.
```

Comment:
Didn't realize this was happening but now that you mention it I think I've ran into this issue before

Reply:
Yeah, this is pretty confusing and took me a while to realize as well. At first I worked around it by setting `dconf.settings."org/postmarketos/megapixels".postprocessor = "${pkgs.megapixels}/share/megapixels/postprocess.sh";` with Home Manager but not everyone using NixOS uses Home Manager so this is not a good fix. A slightly more generic would be to change Megapixels to check if the path is still valid and if it is not, fall back to the default script. This would at least prevent taking no pictures at all but it would still loose the choice made by the user (e.g. using postprocessd).